### PR TITLE
Add retries to sy.login API and notebook polishes

### PR DESCRIPTION
**Pull request by shubham3121**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/grid/client/client.py [hunk 1]
```diff
@@ -101,9 +101,13 @@ def login(
     password: Optional[str] = None,
     conn_type: Type[ClientConnection] = GridHTTPConnection,
     verbose: Optional[bool] = True,
-    timeout: Optional[int] = 120,
+    timeout: Optional[float] = None,
+    retry: Optional[int] = None,
 ) -> Client:
 
+    retry = retry if retry else 5  # Default to 5 retries
```
Comment (rasswanth-s) [inline packages/syft/src/syft/grid/client/client.py:108]: MAYBE: set the default retry and timeout value in argument parameters

File: packages/syft/src/syft/grid/client/client.py [hunk 2]
```diff
@@ -151,13 +155,36 @@ def login(
...
+                credentials=credentials,
+                conn_type=conn_type,
+                timeout=timeout,
+            )
+        except requests.ConnectTimeout:
+            raise requests.ConnectTimeout(
+                f"Connection to node with: {url} timed out. Please try again !!!"
+            )
+        except requests.ConnectionError as e:
+            if retry_attempt < retry:
```
Comment (rasswanth-s) [inline packages/syft/src/syft/grid/client/client.py:175]: Should it be <= , so we try exactly the retry times.

File: packages/syft/src/syft/grid/client/client.py [hunk 3]
```diff
@@ -151,13 +155,36 @@ def login(
...
+    timeout_btw_retries = timeout // retry
+    retry_attempt = 1
+
+    while node is None and retry_attempt < retry:
+        try:
+            node = connect(
+                url=grid_url,
+                credentials=credentials,
+                conn_type=conn_type,
+                timeout=timeout,
```
Comment (rasswanth-s) [inline packages/syft/src/syft/grid/client/client.py:168]: Q: Should we pass in a lower timeout value here, as we pass the default 60 seconds over here?

File: packages/syft/src/syft/grid/client/client.py (hunk 1 above)
Comment (rasswanth-s) [reply packages/syft/src/syft/grid/client/client.py:108]: instead of None

**Review by rasswanth-s — APPROVED** [review]
LGTM, left some comments before merge.

File: packages/syft/src/syft/grid/client/client.py (hunk 1 above)
Comment (shubham3121) [reply packages/syft/src/syft/grid/client/client.py:108]: cause we're using Optional typing which enforces that the value should be None. Therefore setting is later in the code.

File: packages/syft/src/syft/grid/client/client.py (hunk 3 above)
Comment (shubham3121) [reply packages/syft/src/syft/grid/client/client.py:168]: yep, in that case, I will use a lower timeout and set timeout_btw_retries = timeout.

Comment (shubham3121) [reply packages/syft/src/syft/grid/client/client.py:168]: Fixed in https://github.com/OpenMined/PySyft/pull/6578/commits/56b1d0d30d0b66964a8ecfd1be57d47a4d61d2de

File: packages/syft/src/syft/grid/client/client.py (hunk 2 above)
Comment (shubham3121) [reply packages/syft/src/syft/grid/client/client.py]: Fixed in: https://github.com/OpenMined/PySyft/pull/6578/commits/56b1d0d30d0b66964a8ecfd1be57d47a4d61d2de

**Pull request merged**